app/main.py

```diff
@@ -1,6 +1,39 @@
 from fastapi import FastAPI
 from fastapi.middleware.cors import CORSMiddleware
+from pydantic import BaseModel
+import uvicorn
+from langchain_community.document_loaders import CSVLoader
+from langchain_community.vectorstores import Chroma
+from langchain_community.embeddings import HuggingFaceEmbeddings
+import os
+import traceback
+import requests
+import csv
+import math
 
+EMBEDDING_MODEL = HuggingFaceEmbeddings(model_name="sentence-transformers/distiluse-base-multilingual-cased-v2")
+
+GOOGLE_MAP_GEOCODING_API_KEY = os.getenv("GOOGLE_MAP_GEOCODING_API_KEY")
+
+persist_directory = "./app/vectordb"
+
+csv_filepath = "./app/data/example.csv"
+
+fieldlist = ["id", "name", "long", "lat", "area", "category"]
+
+LONG_INDEX = 2
+LAT_INDEX = 3
+
+async def get_cordinates(place_name: str) -> float:
+    url = f"https://maps.googleapis.com/maps/api/geocode/json?address={place_name}&key={GOOGLE_MAP_GEOCODING_API_KEY}"
+    response = requests.get(url)
+    data = response.json()
+    if data['status'] == 'OK':
+        lat = data['results'][0]['geometry']['location']['lat']
+        long = data['results'][0]['geometry']['location']['lng']
+        return lat, long
+    else:
+        return None, None
```

Comment:
前回指摘した点が少し忘れてますね。
ここは`lat`, `lng`の読み取りをするコードを読み込むのに実行時間を費やしてしまうので次のように修正しましょう。
ここで問題です:
```
(None, None)とNoneは別物でしょうか？それとも同じでしょうか？
```
これを間違えるとコードの修正中に予期しないエラーが連発します。配列と辞書以外いらないと思っていたかもしれませんがタプルはめちゃくちゃ使えるのでこのタイミングで一度勉強してみましょう
```suggestion
    if data['status'] != 'OK':
        return None
    lat = data['results'][0]['geometry']['location']['lat']
    long = data['results'][0]['geometry']['location']['lng']
    return lat, long
```

Reply:
> (None, None)とNoneは別物でしょうか？それとも同じでしょうか？

`(None, None)`はタプル型で返ってきてて、`None`は一つだけNoneが返ってくる、みたいな感じ？
別物だと思う！

Coodinate型を返すようにした！
```
    if data['status'] != 'OK':
        return None
    else:
        search_point_coordinate = Coordinate(
            coordinate=(data['results'][0]['geometry']['location']['lat'], data['results'][0]['geometry']['location']['lng'])
        )
        return search_point_coordinate
```